openmdao/core/tests/test_problem.py

```diff
@@ -2044,6 +2044,14 @@ def test_feature_list_problem_vars(self):
                                           'vectorize_derivs',
                                           'cache_linear_solution'])
 
+    def test_out_of_order_set_val(self):
```

Comment:
I think a more explicit test name would be better.  Something like "test_error_msg_set_val_before_setup"